### ISSUE TITLE
Updating Stall speeds.

### ISSUE THREAD

**Issue by DeputyLOL** [ferram4/Ferram-Aerospace-Research#119]
I've noticed that occasionally my air craft can stall without FAR noticing. 
The type of stall where the pitch control surface is at maximum and the aircraft is still descending. FAR only really notices a BIG stall, and by that time its kind of obvious and a bit late.

Would it be possible for FAR to calculate the minimum speed of the aircraft while in flight? This would change in flight with altitude, atm pressure, flap config, gear, AoA and spoilers. I don't know how you would calculate it but knowing my minimum speed could help recover quicker.

If possible, do you think you could make it available for RPM as well?
Thanks

**Comment by ferram4** [comment]
FAR already notices "minor stalls" when 0.005 of the total wing area is stalled; any less than that and it flickers on and off nearly constantly at higher AoAs.  I can lower that, but I expect that it will be less useful as a result.

So as for calculating minimum airspeeds, that's possible, but due to how maximum lift coefficient varies with Mach number, would be *extremely* expensive to calculate in real time.  That assumes that it can find the global maximum as well, since it's possible for small surfaces to stall, while larger ones don't, and so the plane can produce more lift at a higher AoA despite the minor stalling. I'm sorry, I can't justify doing an iterative search for the global maximum of lift every frame and hope to keep performance reasonable.

**Comment by DeputyLOL** [comment]
yeah, I understand where your coming from.

Would it be possible to detect when the plane is about to stall as well as when it is stalling?

**Comment by DeputyLOL** [comment]
Also, sort of unrelated, I love the flaps and spoilers mechanic, but have you considered taking it further with more adjustable control surfaces? Such as slats, speed brakes (different from spoilers) and trim?

**Comment by DeputyLOL** [comment]
You can close this if you want, still want to know about the different control surfaces though... :D

**Comment by ferram4** [comment]
Nah, it'll stay open since this is something that should be there, but needs a bunch of other changes to happen first.  I don't like closing issues unless I've actually resolved them in some way.

As for control surfaces, speed brakes would act pretty much the same way as spoilers, slats are already modelled if you place flaps on the front of a wing, and trim is already part of the stock game.

**Comment by DeputyLOL** [comment]
Speed brakes are different from spoilers as they are deployed in the air to slow the plane down they don't extend as far as spoilers.
Spoilers deploy on the ground and extend pretty much flat. I'm not worried about these now.

As for slats, would it be possible to have control surfaces that physically extend out, instead of rotating.

I am aware about the trim in game, but this isn't what I'm after. I know that this would probably be very difficult. 
Real trim rotates the whole wing but doesn't move the control surface.
![image](https://cloud.githubusercontent.com/assets/12616126/12780485/2b4fa2ac-ca66-11e5-9265-ce99755ffe86.png)

Basically, would you be able to rotate the entire horizontal stabilizer as well as the control surfaces.

**Comment by ferram4** [comment]
On airbrakes, they seem to deploy pretty far on this [MiG-15](http://www.bvmjets.com/turnkey/MiG-Sam/mig1.jpg), and setting up control surfaces in an identical way will provide the same behavior.  In addition, spoilers can deploy in the air and (while not currently set up to be sued as such) are sometimes used for additional roll control and increasing descent rate on final approach for some large jets.

For slats, it is impossible to detect when a control surface is going to be used as a slat or a flap for visual behavior, otherwise a plane with flaps and slats deployed will have them shift orientation and extension should the plane end up moving backwards at any point.  Also, many parts are not modeled suitably for that purpose and would be floating in midair and FAR is not in the business of adding new models.  The behavior is modeled, the visuals will not be.

As for trim, that is *one* of the methods for handling trim, but it is not the only one.  The earliest plane I know of to have that is the Fw 190, but aside from jet airliners, most planes either go with full all-moving stabilizers or else the old-fashioned trim tab on the end of the elevator to keep it auto-deflected.

**Comment by DeputyLOL** [comment]
I am talking in terms of jet airliners. 
As for the slats, there is no need for extra models and this could be easily solved, but this would only work with the movable surfaces like the elevons. Have the surface retract instead of extend, so that the position of it in the editor is the extended position and have the surface move back into the wing when it is retracted. I hope that made sense.

As for trim, would it be possible to have the 'full all-moving stabilizer' in FAR? That was the much better way of describing what I was trying to describe in the previous post. :D

**Comment by ferram4** [comment]
No, you don't understand about the slats.  The behavior is already there.  There is no point in modeling the looks, especially since it either A) shows inconsistent animations based on the movement direction of the vehicle, which makes no logical sense or B) requires more setup from users, and no one is going to make use of it.  I'm not putting extra work in for no actual benefit, and yes, they will still require a new model in order to include the extra connections connecting the slat to the wing.  Otherwise you're just asking for an extendable forward flap, which also isn't happening.

And as for the all-moving stabilizer, that has existed in stock KSP since long before FAR even existed: attach any of the canards / tailfins that have the entire control surface move.  Tada, all moving stabilizer.  Seriously, that's all it is.

**Comment by DeputyLOL** [comment]
... Alright forget the slats.

As for the all moving stabilisers, I know about the stock ones but the current trim system in KSP is not really that useful. I'm designing a jet after the Boeing 737, it flys well but runs out of pitch authority at low speeds and starts to fall (even though it's not stalled)
I was wondering if there was the possibility of a trim system SEPERATE from the control system.
Stock trim works by holding the controls in a certain position, if you trim pitch all the way up you lose the ability to pull up as the control are already maxed. 
If there was a trim that adjusted the stabiliser WITHOUT affect the controls, it could lead to much better handling and much lower speeds, especially with big aircraft. I use the "FAT-455 Aeroplane Tail Fin" as the horizontal stabilisers and they work well. I hope I have explained it better

**Comment by ferram4** [comment]
But ultimately, the trim system is limited by the maximum travel of the control surfaces in real life and there's little reason to think that that's not how any of those systems would behave.  The current trim system behaves approximately as a realistic trim system would behave for one that is directly connected to the control surfaces: move the zero point between the upper and lower limits.  I see no compelling reason to reinvent the wheel, especially considering that it will lead to the strange situation where controls cannot be pitched up fully until they are also trimmed up fully...

**Comment by DeputyLOL** [comment]
I see your reason, but as the trim is separate from the control surface, there is a lot more authority as even if the trim is pitched up fully, the zero point doesn't move. Also, adding realism to this, in real life pilots do need to trim up in-order to pull up.
Ill try and explain with pictures.
![trim](https://cloud.githubusercontent.com/assets/12616126/12799379/6ea83f18-cac6-11e5-8d1f-f99f11da44e6.png)
Blue arrow shows total authority of the stabilizer.

**Comment by blowfishpro** [comment]
So what you're asking for are all-moving control surfaces with additional movement on the trailing edge?  Perhaps this would be useful, but I think it's more complicated to implement in KSP than you realize.

**Comment by DeputyLOL** [comment]
i think that's basically what I'm after, I knew it was going to be difficult. If its too much then we can forget about it.
